tests/testthat/test-read_alevin.R

```diff
@@ -15,17 +15,23 @@ test_that("reading salmon alevin data works", {
   # check that column names are barcodes
   col_barcode <- str_detect(colnames(sce), "^[ACGT]+$")
   expect_true(all(col_barcode))
+  expect_equal(sce@metadata$mapping_tool, "alevin")
+  expect_null(sce@metadata$alevin_fry_version)
 })
 
 test_that("reading alevin-fry USA mode works", {
   sce <- read_alevin(usa_dir,
                      usa_mode = TRUE,
-                    which_counts = "spliced")
+                     which_counts = "spliced")
   expect_s4_class(sce, "SingleCellExperiment")
   expect_equal(dim(sce), sce_af_size)
   # check that column names are barcodes
   col_barcode <- str_detect(colnames(sce), "^[ACGT]+$")
   expect_true(all(col_barcode))
+  # check metadata
+  expect_equal(sce@metadata$mapping_tool, "alevin-fry")
```

Comment:
```suggestion
  expect_false(is.na(sce@metadata$salmon_version))
  expect_false(is.na(sce@metadata$reference_index))
  expect_equal(sce@metadata$mapping_tool, "alevin-fry")
```
Probably a good idea to add checks that the salmon version and reference index aren't empty to ensure the `cmd_info.json` was read in properly.

Reply:
yup, good idea... made me realize the line after is wrong though: should be `is.null`!

Reply:
Oh, actually, I know why I wasn't testing the reference index: it would fail with some versions where that data wasn't reported properly... so null values there are possible. (this was also why I wasn't testing for salmon version probably. But that is there (if sometimes wrong) in all cases.